backend/test/resolver.spec.js

```diff
@@ -1,91 +1,220 @@
+    mutation login($email: String!, $password:String!) {
+        login(email: $email, password: $password) {
+            token
+            user
         }
     }
 `;
 
-const CREATE_ENTRY = gql`
-    mutation createEntry($text: String!) {
-        createEntry(text: $text) {
-            id,
-            text,
-            editMode
+const ALL_TODOS_QUERY = gql`
+    query todosQuery{
+        todos{
+            id
+            text
+            user {
+                name
+            }
         }
     }
 `;
 
-const DELETE_ENTRY = gql`
-    mutation DeleteEntry($id: String!) {
-        deleteEntry(id: $id)
+const ALL_TODOS_QUERY_PAGINATED = gql`
+    query todosQuery ($page: Int, $size: Int ){
+        todos (page: $page, size: $size){
+            id
+            text
+            user {
+                name
+            }
+        }
     }
 `;
 
-const LOGIN = gql`
-    mutation login($email: String!, $password:String!) {
-        login(email: $email, password: $password)
+const CREATE_TODO = gql`
+    mutation createTodo($text: String!) {
+        createTodo(text: $text) {
+            id
+            text
+            user {
+                name
+            }
+        }
+    }
+`;
+
+const UPDATE_TODO = gql`
+    mutation updateTodo($id: String!, $text: String!){
+        updateTodo(id: $id, text: $text){
+            id
+            text
+        }
+    }
+`;
+
+const DELETE_TODO = gql`
+    mutation DeleteTodo($id: String!) {
+        deleteTodo(id: $id)
     }
 `;
 
 describe('User is logged in', () => {
+    beforeEach(async () => {
+        const result = await testClient.mutate({
+            mutation: LOGIN,
+            variables: {email: testUserEmail, password: testUserPassword}
+        });
+        let req = new Map();
+        req.set("Authorization", result.data.login.token);
+        testServer = getTestApolloServer(req);
+        testClient = createTestClient(testServer);
+        query = testClient.query;
+        mutate = testClient.mutate;
+    });
+
+    afterEach(async () => {
+        //remove token header
+        testServer = getTestApolloServer();
+        testClient = createTestClient(testServer);
+    });
+
     describe('Queries', () => {
-        it("has start todo items", () => {
-            query({ query: ALL_ITEMS_QUERY }).then((data) => {
-                expect(data.data.items).toHaveLength(4);
-                expect(data.data.items).toMatchObject([
-                    { id: 1 },
-                    { id: 2 },
-                    { id: 3 },
-                    { id: 4 }
-                ]);
-            });
+        it("has start todo todos", async () => {
+            const data = await query({query: ALL_TODOS_QUERY});
+            expect(data.errors).toBeUndefined();
+            expect(data.data.todos.length).toBeGreaterThan(0);
+            expect(data.data.todos.length).toBeLessThanOrEqual(20) // 20 is the std
+            expect(data.data.todos[0].user.name).toBe(testUserName)
```

Comment:
I would definitely merge those expectations into one:
```js
await expect(query({query: ALL_TODOS_QUERY})).resolves.toMatchObject({
  data: {
    todos: [
      { user: { name: 'testuser' }
    ]
  },
  errors: undefined,
})
```

And please hard-code your expected value, don't use a variable `testUserName` which the code reviewer must lookup first to understand the test case.